components/greaselion/browser/greaselion_download_service.cc

```diff
@@ -140,6 +150,18 @@ bool GreaselionRule::Matches(GreaselionFeatures state) const {
   if (!PreconditionFulfilled(preconditions_.auto_contribution_enabled,
                              state[greaselion::AUTO_CONTRIBUTION]))
     return false;
+  if (!PreconditionFulfilled(preconditions_.supports_minimum_browser_version,
+                          state[greaselion::SUPPORTS_MINIMUM_BROWSER_VERSION]))
+    return false;
+  // Validate against browser version.
+  if (!minimum_brave_version_.empty()) {
+    bool rule_version_is_higher_than_browser =
+        (browser_version.CompareToWildcardString(minimum_brave_version_) < 0);
```

Comment:
Looks like this might DCHECK if `minimum_brave_version_` is not a valid wildcard:

https://source.chromium.org/chromium/chromium/src/+/master:base/version.cc;l=111;drc=58e017c522e627f8b2765bfbb244d7fbafc79710

Reply:
agreed. The `if` clause should call `Version::IsValidWildcardString` instead of `.empty()`

Reply:
Resolved here and added tests to verify https://github.com/brave/brave-core/compare/9ae3a2848bde04e068fe958d28508882c44fa299..8e11111c2d631ecdd0a3b224278f6fbde12febf1